express-notes-api/index.js

```diff
@@ -0,0 +1,101 @@
+const dataJSON = require('./data.json');
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+app.get('/api/notes', (req, res) => {
+  const notesArr = [];
+  for (const prop in dataJSON.notes) {
+    notesArr.push(dataJSON.notes[prop]);
+  }
+  res.status(200).json(notesArr);
+
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const noteId = Number(req.params.id);
+
+  if (noteId <= 0) {
+    res.status(400).json({ error: 'noteId must be a positive integer' });
+  } else if (!dataJSON.notes[noteId]) {
+    res.status(404).json({ error: 'Cannot find note with id ' + noteId });
+  } else if (dataJSON.notes[noteId]) {
```

Comment:
Just an `else` here would be better, you already checked if the note exists above. There is no point to check the opposite of what was already checked, that is what an `else` already does.